src/fs.c

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "fs.h"
+#include "string.h"
+#include "kerneltypes.h"
+#include "process.h"
+#include "iso.h"
+#include "console.h"
+#include "sys_fs_err.h"
+
+#define READ 4
+#define WRITE 2
+#define APPEND 1
+
+uint32_t fs_security_check(const char *path);
+
+#define MAX_OS_OPEN_FILES 1024
+
+struct fs_agnostic_file open_files_table[MAX_OS_OPEN_FILES];
+
+/*
+ * A dummy function to help us in our debugging
+ */
+void fs_print_allowances() {
+    struct list_node *iterator = current->fs_allowances_list.head;
+    while (iterator) {
+        console_printf("%s -> ", ((struct fs_allowance *)iterator)->path);
+        iterator = iterator->next;
+    }
+    console_printf("\n");
+}
+
+void fs_sys_init_open_files_table() {
+    int i;
+    for (i = 0; i < MAX_OS_OPEN_FILES; i++) {
+        open_files_table[i].filep = 0;  // Set each file pointer to null
+    }
+}
+
+void fs_init_security(struct process *p) {
+    // Initialize all file descriptor slots to 0
+    int i;
+    for (i = 0; i < PROCESS_MAX_OPEN_FILES; i++) {
+        p->fd_table[i].is_open = 0;
+        p->fd_table[i].ptr = 0;
+    }
+
+    //Allowances alloc'ing.
+    struct fs_allowance *allowance = kmalloc(sizeof(*allowance));
+    strcpy(allowance->path, "/");
+    allowance->do_allow_below = 1;
+
+    //Set the lone permission in the list to root (that is all of the fs)
+    list_push_head(&(p->fs_allowances_list), (struct list_node *)allowance);
+}
+
+// Look at the OS's table of open files to see if read write conflicts
+// might be occurring amongst many processes if you allow it to be opened
+bool fs_check_open_conflict(const char *path, const char *mode) {
+    return 1;
+}
+
+int map_media_to_driver_id(int media) {
+    return ISO;
+}
+
+int mode_str_to_int(const char *mode) {
+    if (strcmp(mode, "a") == 0) {
```

Comment:
If we had a `contains(char c)` function in `string.h` I would say yes, but this looks fine to me, especially because this is every possible case already enumerated.